### ISSUE TITLE
Bug: [no-confusing-void-expression] Wrong autofix for `a ? void : T`

### ISSUE THREAD

**Issue by liuxingbaoyu** [typescript-eslint/typescript-eslint#6187]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&code=MYewdgzgLgBAtgQwJ4CMCmBRATllDgDWMAvDABQCUJAfDAN4CwAUDDADZqwIkwCMA3M1ZZOAVyxgY3APzkAZqLDAoAS3DkqdGAF8KlGAC4pgptpPNmaAB4AHEFlgATNHISi2sOtqA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYkACAXAngA4oDG0AlobgLQrzkB2uA9AwPbUlsMBmsyjAObUAbm3IATWgA9C0FAO7ooiaNDbRIAXxBagA&tsconfig=N4XyA

### Repro Code

```typescript
const maybeErrback = () => {
  let a = 1;
  return a ? (function () { })() : a;
};


export default {}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {"@typescript-eslint/no-confusing-void-expression": "error"}
}
```

### tsconfig

```jsonc
{}
```

### Expected Result

Don't report it or disable automatic fixes.

### Actual Result

was wrongly fixed.

```ts
const maybeErrback = () => {
  let a = 1;
  a ? (function () { })() : a;
};


export default {}
```
